packages/wasm/src/keys.ts

```diff
@@ -28,6 +32,23 @@ export const getEphemeralByIndex = (fullViewingKey: string, index: number): Addr
   return new Address({ inner: uintArray });
 };
 
+export const getIndexByAddress = (fullViewingKey: string, address: string): AddressIndex => {
+  const res = validateSchema(
+    z.object({
+      account: z.number().optional(),
+      randomizer: z.string(),
+    }),
+    is_controlled_address(fullViewingKey, address),
```

Comment:
Is appears to return a `Option<pb::AddressIndex>`. Does this mean that the result could potentially be undefined? If so, that zod type is not right.

Reply:
if account = 0, then return only randomizer

Reply:
But let's say you pass an address into `is_controlled_address` that isn't connected with that fullViewingKey, does it return undefined?

Reply:
res as undefined

Reply:
i see, you mean is_controlled_address can return undefined

Reply:
what we should return if is_controlled_address return undefined?

Reply:
throw new Error('no address given in request')?